src/power_grid_model/utils.py

```diff
@@ -54,6 +55,35 @@ def _get_component_scenario(component_scenarios: BatchArray) -> np.ndarray:
     return {component: _get_component_scenario(component_data) for component, component_data in dataset.items()}
 
 
+def get_data_set_batch_size(dataset: BatchDataset) -> int:
```

Comment:
i know it's linguistically incorrect but across the code base we use
```suggestion
def get_dataset_batch_size(dataset: BatchDataset) -> int:
```